addons/wowl/static/tests/components/dropdown_tests.ts

```diff
@@ -0,0 +1,475 @@
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("multi-level dropdown: parent closing modes on item selection", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown, DropdownItem };
+    static template = owl.tags.xml`
+      <Dropdown>
+        <t t-set-slot="menu">
+          <Dropdown>
+            <t t-set-slot="menu">
+              <DropdownItem class="item1" parentClosingMode="'none'" />
+              <DropdownItem class="item2" parentClosingMode="'closest'" />
+              <DropdownItem class="item3" parentClosingMode="'all'" />
+              <DropdownItem class="item4" />
+            </t>
+          </Dropdown>
+        </t>
+      </Dropdown>
+    `;
+  }
+  parent = await mount(Parent, { env });
+
+  // Open the 2-level dropdowns
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 2);
+
+  // Select item (parentClosingMode=none)
+  await click(parent.el!, "li.item1");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 2);
+
+  // Select item (parentClosingMode=closest)
+  await click(parent.el!, "li.item2");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 1);
+
+  // Reopen second level dropdown
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 2);
+
+  // Select item (parentClosingMode=all)
+  await click(parent.el!, "li.item3");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+
+  // Reopen the 2-level dropdowns
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 2);
+
+  // Select item (default should be parentClosingMode=all)
+  await click(parent.el!, "li.item4");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("multi-level dropdown: payload bubbles on item selection", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown, DropdownItem };
+    static template = owl.tags.xml`
+      <Dropdown t-on-dropdown-item-selected="onItemSelected">
+        <t t-set-slot="menu">
+          <Dropdown t-on-dropdown-item-selected="onItemSelected">
+            <t t-set-slot="menu">
+              <DropdownItem payload="{ answer: 42 }" />
+            </t>
+          </Dropdown>
+        </t>
+      </Dropdown>
+    `;
+    onItemSelected(ev: OwlEvent<any>) {
+      assert.deepEqual(ev.detail.payload, { answer: 42 });
+    }
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+
+  // As two listeners are defined in the template,
+  // clicking once the item would execute the handler twice.
+  assert.expect(2);
```

Comment:
typically, we define this as first line of the test

Reply:
:+1: